[achl1012] iP

achl1012 opened the pull request:
### Charlotte

> "Your mind is for having ideas, not holding them." – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Charlotte frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [ ] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ] Reminders (coming soon)


If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public static void main(String[] args) {
    new Charlotte("data/charlotte.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
rename file, add greeting and exit

Commit:
Echoes commands

Commit:
Add items and show list

Commit:
Add Task.java and mark as done feature

Commit:
Add ToDos, Events, Deadlines

Commit:
Handle errors

Commit:
Add delete tasks feature

Commit:
Automated text UI testing

Commit:
Save the tasks in the hard disk

Commit:
Merge Branch-Level-7 to master

Commit:
Merge branch 'master' of https://github.com/achl1012/ip

Commit:
Understand dates and times

Commit:
Merge branch 'branch-Level-8'

Commit:
Make the code more OOP

Commit:
Divide classes into packages

Commit:
Merge branch 'add-gradle-support'

Commit:
Add JUnit tests

Commit:
Package the app as a JAR file

Commit:
Add JavaDoc comments

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Follow Java coding standard

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Find a task using a keyword

Commit:
Find a task using a keyword

Commit:
Merge branch 'branch-Level-9'

tanyqcollin commented on src/main/java/charlotte/storage/Storage.java:
```diff
@@ -0,0 +1,127 @@
+                        break;
+                    default:
+                        throw new CharlotteException("Unknown task type in file: " + taskType);
+                }
```
The `case` clauses should have same indentation to the `switch`
eg.
```
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}
```

tanyqcollin commented on src/main/java/charlotte/command/AddCommand.java:
```diff
@@ -0,0 +1,52 @@
+     * @param storage The Storage object used to save the updated task list.
+     * @throws CharlotteException If an error occurs while saving the task list to storage.
+     */
+    @Override
```
nice javadoc comments, complete and adheres to coding guidelines

tanyqcollin commented on src/main/java/charlotte/command/Command.java:
```diff
@@ -0,0 +1,40 @@
+     * @throws CharlotteException If an error occurs during the execution of the command.
+     */
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws CharlotteException;
+
```
Appropriate use of abstract class based on abstract method.

tanyqcollin commented on src/main/java/charlotte/storage/Storage.java:
```diff
@@ -0,0 +1,127 @@
+public class Storage {
+    private final String filePath;
+    private static final ArrayList<Task> TASKS = new ArrayList<>();
+
```
good use of final and static variable

tanyqcollin reviewed (commented):
LGTM. The code looks good for me and does not have any major issues. Javadoc are documented precisely and completed.

Commit:
Use checkStyle to detect coding style violations

Commit:
Use checkStyle to detect coding violations

Commit:
Merge branch 'branch-A-CheckStyle'

volleyballkickedme commented on src/main/java/charlotte/command/FindCommand.java:
```diff
@@ -6,13 +6,29 @@
+     * @throws CharlotteException If an error occurs during the execution.
+     */
     @Override
     public void execute(TaskList tasks, Ui ui, Storage storage) throws CharlotteException {
```
good adherence to coding standards

volleyballkickedme commented on src/main/java/charlotte/command/FindCommand.java:
```diff
@@ -6,13 +6,29 @@
         this.keyword = keyword;
     }
 
+    /**
```
good use of javadoc to explain method. I had no problems understanding the code here

Commit:
Add a GUI to the chatbot

Commit:
Add a GUI to the chatbot

Commit:
Merge branch 'branch-A-Level-10'

Commit:
Use assertions

Commit:
Use assertions

Commit:
Improve code quality

Both the parse method in the Parser class and the parseTask method in the Storage class contain case statements that have some common behaviours and the methods are longer than 30 LoC.

The common behaviours across the case statements causes code duplication and long methods affect readability of code.

Extracting the common behaviour into helper methods helps to make each method shorter and improve readability.

Let's create helper methods to handle common behaviour in each method.

This extracts out the common logic and makes each method shorter and more readable.

achl1012 committed:
Merge pull request #1 from achl1012/branch-A-Assertions

Add assertions to the code

Commit:
Merge branch 'master' of https://github.com/achl1012/ip

Commit:
Merge branch 'master' into branch-A-CodeQuality

achl1012 committed:
Merge pull request #2 from achl1012/branch-A-CodeQuality

Improve code quality

Commit:
Merge branch 'master' of https://github.com/achl1012/ip

Commit:
Add reminders for tasks

There is no feature for users to get reminders on upcoming tasks in their task list.

Providing a way for users to get reminders about tasks will improve user experience and make it easier for them to view upcoming tasks.

Let's provide a way for users to get reminded about upcoming tasks using the "reminders" command.

Users can easily view the list of tasks when they wish to be reminded of upcoming deadlines or events starting within one week.

Commit:
Merge branch 'branch-B-Reminders'

Commit:
Improve the GUI

Commit:
Give a unique personality

Commit:
Improve code quality

Commit:
Add a screenshot of the product

Commit:
Add a User Guide

Commit:
Amend user guide

Commit:
Amend user guide
